prefq/server.py

```diff
@@ -55,14 +61,53 @@
 DEFAULT_DEBUG = False
 
 
-def before_first_request():
+def before_first_request(sshkey):
     """Define starting routine"""
 
+    if sshkey is not None:
+        with open(sshkey, "rb") as key_file:
+            app.config["PUBLIC_KEY"] = serialization.load_ssh_public_key(
+                key_file.read(),
+            )
+
     # Create video folder (if necessary)
     if not os.path.exists(app.config["VIDEO_FOLDER"]):
         os.mkdir(app.config["VIDEO_FOLDER"])
 
 
+def encrypt(message):
+    """Encrypt message with public key"""
+
+    encrypted_message = app.config["PUBLIC_KEY"].encrypt(
+        message.encode(),
+        padding.OAEP(
+            mgf=padding.MGF1(algorithm=hashes.SHA256()),
+            algorithm=hashes.SHA256(),
+            label=None,
+        ),
+    )
+
+    encrypted_message = base64.b64encode(encrypted_message).decode("utf8")
+    return encrypted_message
+
+
+@app.route("/challenge", methods=["GET"])
+def send_challenge():
+    """
+    Send a challenge to the Query Client
+
+    This is done in order to ensure secure communication.
+
+    The Query Client has to decrypt the challenge with the private key,
+    and send it back to the server, in order to prove its identity.
+    """
+
+    password = secrets.token_urlsafe(32)
+    app.config["SERVER_PW"] = password
```

Comment:
I think it would be better to call this `SERVER_CHALLENGE_TOKEN_PLAIN` (and `password` -> `random_challenge_token`).